BSPConvert.Lib/Source/ShaderParser.cs

```diff
@@ -357,9 +357,16 @@ private void ParseRGBGen(ShaderStage stage, string[] split)
 			switch (split[1].ToLower())
 			{
 				case "wave":
-					// TODO: Parse wave form
-					stage.rgbGen = ColorGen.CGEN_WAVEFORM;
-					break;
+					{
+						stage.rgbGen = ColorGen.CGEN_WAVEFORM;
+						
+						stage.rgbWave.func = NameToGenFunc(split[2]);
```

Comment:
Refactor this into a method that passes in an `ArraySegment` from index 2 to 6 and returns a `WaveForm`. If you're not sure how to use `ArraySegment`, check the `ParseVector` method that I added recently.